flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -1,29 +1,202 @@
+            result(FlutterMethodNotImplemented)
+        }
+
+
+    }
+
+    private func initNativeSdk(_ call: FlutterMethodCall, result: @escaping FlutterResult){
+        guard let arguments = call.arguments as? [String: Any] else {
+            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil) )
+            return
+        }
+
+        if (arguments.isEmpty) {
+            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil) )
+            return
+        }
+
+        SentrySDK.start { options in
+            if(arguments["dsn"] != nil ){
+                options.dsn = arguments["dsn"] as? String
+            } else {
+                result(FlutterError(code: "4", message: "A valid Dsn must be provided", details: nil) )
+            }
+            if let isDebug = arguments["debug"] as? Bool {
+                options.debug = isDebug
+            }
+
+            if let environment = arguments["environment"] as? String{
+                options.environment = environment
+            }
+            if let releaseName = arguments["release"] as? String{
+                options.releaseName = releaseName
+            }
+
+            if let enableAutoSessionTracking = arguments["enableAutoSessionTracking"] as? Bool {
+                options.enableAutoSessionTracking = enableAutoSessionTracking
+            }
+
+            if let attachStacktrace = arguments["attachStacktrace"] as? Bool {
+                print("attachStacktrace \(attachStacktrace)")
+                options.attachStacktrace = attachStacktrace
+            }
+
+            if let sessionTrackingIntervalMillis = arguments["sessionTrackingIntervalMillis"] as? UInt{
+                options.sessionTrackingIntervalMillis = sessionTrackingIntervalMillis
+            }
+
+            if let dist = arguments["dist"] as? String{
+                options.dist = dist
+            }
+
+            if let integrations = arguments["integrations"] as? [String]{
+                options.integrations?.append(contentsOf: integrations)
+            }
+
+            if let enableAutoNativeBreadcrumbs = arguments["enableAutoNativeBreadcrumbs"] as? Bool,
+               enableAutoNativeBreadcrumbs == false {
+                options.integrations = options.integrations?.filter { (name) -> Bool in
+                    return name != "SentryAutoBreadcrumbTrackingIntegration"
+                }
+            }
+
+            if let maxBreadcrumbs = arguments["maxBreadcrumbs"] as? UInt{
+                options.maxBreadcrumbs = maxBreadcrumbs
+            }
+
+            /*
+             missing fields : enableAutoNativeBreadcrumbs, diagnostic level,
+             enableNativeCrashHandling,platform, web, packages,anrTimeoutIntervalMillis
+             diagnosticLevel, cacheDirSize
+             */
+
+            options.beforeSend = { event in
+                self.setEventOriginTag(event: event)
+                /// TODO ? addPackages
```

Comment:
`event.sdk.packages`, we need to merge them

Reply:
I'm not sure sentry-cocoa handle `event.sdk.packages` WDYT ? https://github.com/getsentry/sentry-cocoa/blob/a339a4459442238876e502a8830bf6e628169098/Sources/Sentry/Public/SentryEvent.h#L91-L103

Reply:
@rxlabz not by default because cocoa is a single package, but you can set the packages item to the dictionary yourself.

the idea is:

https://develop.sentry.dev/sdk/event-payloads/sdk/#example

Reply:
✅